unit_tests/Common.d

```diff
@@ -66,10 +85,132 @@ C code translated to:
 Expected D code:
 %1$s
 %3$s
-%1$s/";
+%1$s
+AST dump:
+%4$s/";
 
-    string message = format(fmt, "----------------", translated, d);
+    if (translated != expected)
+    {
+        size_t maxSubmessageLength = 10_000;
+        string astDump = unit.dumpAST(true);
+
+        if (maxSubmessageLength < translated.length)
+            translated = translated[0..maxSubmessageLength] ~ "...";
+
+        if (maxSubmessageLength < astDump.length)
+            astDump = astDump[0..maxSubmessageLength] ~ "...";
 
-    if (translated != d) 
+        string message = format(fmt, sep, translated, expected, astDump);
         throw new TranslateAssertError(message, file, line);
+    }
+}
+
+void assertTranslates(string c, string d, string file = __FILE__, size_t line = __LINE__)
+{
+    auto unit = makeTranslationUnit(c);
+    assertTranslates(d, unit, true, Language.c, file, line);
+}
+
+void assertTranslatesFile(
+    string expectedPath,
+    string objCPath,
+    bool strict,
+    Language language,
+    string[] arguments,
+    string file = __FILE__,
+    size_t line = __LINE__)
+{
+    string readFile(string path) 
+    {
+        import std.stdio : File;
+        auto file = File(path, "r");
+        char[] buffer = new char[file.size];
+        return file.rawRead(buffer).idup;
+    }
+
+    auto expected = readFile(expectedPath);
+    auto unit = TranslationUnit.parse(index, objCPath, arguments);
+    assertTranslates(expected, unit, strict, language, file, line);
+}
+
+string findGNUStepIncludePath()
+{
+    import std.file : isDir, exists;
+    import std.format : format;
+
+    string path = "/usr/include/GNUstep";
+
+    if (exists(path) && isDir(path))
+        return "-I%s".format(path);
+    else
+        return null;
+}
+
+string[] findCcIncludePaths()
+{
+    string[] extract(string output)
+    {
+        import std.algorithm.searching;
+        import std.algorithm.iteration;
+        import std.string;
+
+        string start = "#include <...> search starts here:";
+        string stop = "End of search list.";
+        auto paths = output.findSplitAfter(start)[1].findSplitBefore(stop)[0].strip();
+        return paths.empty ? null : map!(a => "-I%s".format(a))(paths.splitLines()).array;
+    }
+
+    import std.process : executeShell;
+    auto result = executeShell("cc -E -v - < /dev/null");
```

Comment:
The situation with the standard includes is real mess : (. If I do not add that search directories I get complaints like:
` /usr/include/GNUstep/GNUstepBase/GSConfig.h:376:10: fatal error: 'stdbool.h' file not found`

And even when I added that files to `resources` I began to get complaints about headers from `objc` subdirectory (seems objc runtime e.g.`objc/objc.h`) which I could find only in this location on my system:
`/usr/lib/gcc/x86_64-linux-gnu/4.8/include/objc/objc.h`
Despite the fact I have the standalone clang installation.